cli/mfg.c

```diff
@@ -66,7 +66,9 @@ static const struct argconfig_choice secure_state_choices[] = {
 
 static char *spi_rate_str[] = {
 	"100", "67", "50", "40", "33.33", "28.57",
-	"25", "22.22", "20", "18.18"
+	"25", "22.22", "20", "18.18",
+	"120", "80", "60", "48", "40", "34",
+	"30", "26.67", "24", "21.82"
```

Comment:
This is still ugly, but it's the existing code that is ugly.

This array is highly questionably and shouldn't even be in the CLI. I still say change it to a float and pass the actual rate instead of making it even more ugly by having a special offset per generation.

Reply:
Float is much harder to work with... e.g., ``` if (rate < 21.8201 && rate > 21.8199)``` vs ``` if (rate == SWITCHTEC_SPI_RATE_21_82)```

Plus it gives user the impression that all 0.00 - inf.xxx rates are available when in fact only a few rate levels are supported.

Reply:
We don't have any code that's doing comparisons on the rate, it's really only there to be printed or for basic information.

But it's even harder to do comparisons with the current scheme when the user have to map it to an array of strings and then convert it to an integer.

I don't think the user would assume that any valid float could be used for the rate just because it takes a float in. That's not really sensible.

The config_set function already looks pretty broken but nobody uses it, as it requires the user to know the weird mapping which was just changed by this patch.

Use a float, have the config_set find the nearest value in the table to set it to and have config_get set the float from the table. Then you can have a different table for GEN4 and GEN5 in the library where it belongs.

Reply:
If you really don't like floats then use an int specified in KHz or some such.... But you still need to do a find nearest operation in confg_set, so it doesn't really get you anything except making it harder to print in the CLI.

Reply:
@lsgunth Printing `float` is tricky ('22.22' printed as `21.19999`) and is hard to maintain compatibility with output of earlier versions. So the code has been updated to use `int` for clock rate in KHz. 

Considering that GEN4 has already been deployed, we do the following to maintain compatibility with earlier versions:
- at API level, we add a new set of `xxxx_gen5` functions and keep the exist functions as-is
- at CLI level, we are free to use the new API functions, but needs to keep output compatible with earlier versions

Please review and give your comments.